### PR TITLE
Using bitboards

### PR DESCRIPTION
**Pull request by forki**
This improves perf yet another time. Surprisingly the bit fiddeling also works on fable. yay @alfonsogarciacaro

### CODE REVIEW

**Comment by markpattison** [comment]
I can't see what the problem is, but running the tests with this PR, MCTS-200 is losing to both Minimax-2 and MCTS-25, which it wasn't before.

**Comment by forki** [comment]
yeah. please don't merge

File: src/Reversi/Board.fs [hunk 1]
```diff
@@ -1,34 +1,52 @@
...
+        board ||| (1UL <<< pos)
+
+    let inline removeStone (pos:int) (board:uint64) =
+        board &&& (~~~ (1UL <<< pos))
+
+    let inline isSet (pos:int) (board:uint64) =
+        ((board >>> pos ) &&& 1UL) = 1UL
+
+    let countStones (board:uint64) =
+        let mutable count = 0UL
```
Comment (markpattison) [inline src/Reversi/Board.fs]: Should return an int so we can safely subtract values later (this broke the minimax player.)

File: src/Reversi/Computer/MCTS.fs [hunk 2]
```diff
@@ -76,18 +76,23 @@ and Node(parent:Node option,random:Random,board:Board) =
...
-                        let opposite = currentBoard.NextToMove.opposite
+                        let opposite = currentBoard.NextToMove.Opposite
                         currentBoard <- { currentBoard with NextToMove = opposite }
                     else
-                        match Board.getStatus currentBoard with
-                        | Win White -> this.BackProp 1.
-                        | Win Black -> this.BackProp -1.
-                        | Tie -> this.BackProp 0.5
+                        let numBlack = Bitwise.countStones board.BlackSquares
+                        let numWhite = Bitwise.countStones board.WhiteSquares
```
Comment (markpattison) [inline src/Reversi/Computer/MCTS.fs]: Should be `currentBoard` instead of `board` on both lines - this was the problem with the MCTS player.

**Review by markpattison — COMMENTED** [review]
Seems to work fine with these two issues addressed.

**Comment by forki** [comment]
Wow. Thanks

**Pull request merged**